frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -222,20 +234,17 @@ impl<T: Config> SignedSubmissions<T> {
 		})
 	}
 
+	/// Remove the signed submission with the highest score from the set.
+	pub fn pop_last(&mut self) -> Option<SignedSubmissionOf<T>> {
+		let best_index = self.indices.len().saturating_sub(1);
+		self.swap_out_submission(best_index, None)
+	}
+
 	/// Iterate through the set of signed submissions in order of increasing score.
 	pub fn iter(&self) -> impl '_ + Iterator<Item = SignedSubmissionOf<T>> {
-		self.indices.iter().filter_map(move |(_score, &idx)| {
-			let maybe_submission = self.get_submission(idx);
-			if maybe_submission.is_none() {
-				log!(
-					error,
-					"SignedSubmissions internal state is invalid (idx {}); \
-					there is a logic error in code handling signed solution submissions",
-					idx,
-				)
-			}
-			maybe_submission
-		})
+		self.indices
+			.iter()
+			.filter_map(move |(_score, _bn, idx)| self.get_submission(*idx).defensive())
```

Comment:
not sure what `defensive` does here but isn't the log needed anymore?

Reply:
defensive exactly does that :) panics on test, logs on production, assuming it should never happen.